### PR TITLE
[2/Q] Add servlet functionality for opportunity update.

### PR DESCRIPTION
**Pull request by nikhitashanker**
- Add servlet and database wrapper methods for volunteering opportunity update.
- Make naming for volunteering opportunity variables consistent with volunteering opportunity class.

### CODE REVIEW

File: project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java [hunk 1]
```diff
@@ -14,34 +14,57 @@
 /** Servlet that handles submission of the form for creating volunteering opportunities. */
 @WebServlet("/volunteering-form-handler")
 public class VolunteeringFormHandlerServlet extends HttpServlet {
+  private static final String OPPORTUNITY_ID = "opportunity-id";
   private static final String OPPORTUNITY_NAME = "opportunity-name";
   private static final String OPPORTUNITY_NUM_SPOTS = "opportunity-num-spots";
   private static final String SKILL = "skill";
```
Comment (dash102) [inline project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java:20]: For consistency, if this is used only in the context of volunteering opportunities, also name it OPPORTUNITY_X?

File: project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java [hunk 2]
```diff
@@ -14,34 +14,57 @@
...
 
     response.sendRedirect("/events-feed.html");
   }
 
-  private static void insertVolunteeringOpportunityIntoDB(
-      String volunteerName, Integer volunteerNumber, Set<String> requiredSkills) {
-    DatabaseWrapper dbWrapper =
-        new DatabaseWrapper(DatabaseConstants.INSTANCE_ID, DatabaseConstants.DATABASE_ID);
+  private static void insertVolunteeringOpportunityInDB(
+      String opportunityName, int opportunityNumberOfSpots, Set<String> requiredSkills) {
```
Comment (dash102) [inline project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java:46]: Ditto, opportunityRequiredSkills if it's only used for opportunities for consistency

File: project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java [hunk 3]
```diff
@@ -102,10 +102,31 @@ public void insertVolunteeringOpportunity(VolunteeringOpportunity opportunity) {
...
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
+
+    List<Mutation> mutations =
+        getMutationsFromBuilder(newUpdateBuilderFromVolunteeringOpportunity(), opportunity);
+    dbClient.write(mutations);
+    spanner.close();
+  }
+
   private static Mutation.WriteBuilder newInsertBuilderFromVolunteeringOpportunity() {
```
Comment (dash102) [inline project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java:122]: Suggestion: There's also InsertOrUpdate for mutation builders, if that would suit your code better

Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java:122]: Ack, noted but does not work well if primary key is not one of form fields - thanks.

File: project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java [hunk 4]
```diff
@@ -14,34 +14,57 @@
...
         CommonUtils.getParameter(request, OPPORTUNITY_NAME, /* DefaultValue= */ "");
-    Integer volunteerNumber =
+    int opportunityNumberOfSpots =
         Integer.parseInt(
             CommonUtils.getParameter(request, OPPORTUNITY_NUM_SPOTS, /* DefaultValue= */ "0"));
     Set<String> skills = CommonUtils.getParameterValues(request, SKILL);
 
-    insertVolunteeringOpportunityIntoDB(volunteerName, volunteerNumber, skills);
+    // If opportunityId is not passed as a parameter, perform an insert else perform an update
+    if (opportunityId == null)
```
[the remark this answers is not in the document]
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java:36]: Done - missed this, thank you.

File: project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java (hunk 1 above)
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java]: Done - yes, I noticed now that my naming is a bit consistent. I have made it match all variables in the opportunity class across all files now.

File: project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java (hunk 2 above)
Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java]: Done, thanks - made naming consistent.

File: project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java [hunk 5]
```diff
@@ -14,34 +14,50 @@
...
-  private static final String SKILL = "skill";
+  private static final String OPPORTUNITY_ID = "opportunity-id";
+  private static final String NAME = "name";
+  private static final String NUM_SPOTS_LEFT = "num-spots-left";
+  private static final String REQUIRED_SKILL = "required-skill";
   private static final String HARDCODED_EVENT_ID = "0883de79-17d7-49a3-a866-dbd5135062a8";
+  private static final DatabaseWrapper databaseWrapper =
+      new DatabaseWrapper(DatabaseConstants.INSTANCE_ID, DatabaseConstants.DATABASE_ID);
 
   @Override
```
Comment (liangqigong) [inline project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java:25]: Also add comment for the function.

Comment (nikhitashanker) [reply project/src/main/java/com/google/sps/servlets/VolunteeringFormHandlerServlet.java:34]: Done.

**Pull request merged**